### ISSUE TITLE
Applicant dashboard status changes to conditionally approved

### ISSUE THREAD

**Issue by MarsRomer** [bcgov/CONN-CCBC-portal#1338]
As an analyst
I want to be able to decide when the applicant sees their application becomes 'Conditionally approved' on their dashboard
So that I can follow the business process of informing them by email and receiving confirmation before the dashboard shows 'Conditionally approved'

**Description:**
NWBC would like to make sure that the applicant gets the letter of Conditional approval and the applicant agrees to the conditional approval before they see the status on their dashboard change to 'Conditionally Approved'

# Acceptance Criteria:
 [ ] 'Status that the applicant sees' dropdown is disabled until the following 3 criteria are met: BC Ministers Decision = 'Approved'  AND Applicants Response = Agree AND Analyst Status = Conditionally Approved 
<s>- [ ] If analyst changes "Status that applicant sees" and presses save then a modal appears</s>
<s>- [ ] When analyst clicks "Yes, change it" in the modal, then the status changes on the applicant's dashboard.</s>
- [x] Changes made to the 'Status that the applicant sees'  only effects the external status, not the internal status that the analyst sees.  This includes changing the status to Conditionally Approved and  back to Received and any other future options.
<s>- [ ] Analyst can change Status that applicant sees from Conditionally Approved to Received </s>
- [x] The status that the applicant sees on their dashboard is updated when analyst changes it successfully (after modal) 
- [ ] If user clicks 'Cancel' on modals then it cancels the save action, and takes the user back to the edit mode for the conditionally approved.  Keeps the data they had filled in previously the same

**Screen**

![image.png](https://images.zenhubusercontent.com/61ae5dab07758299775d9d78/6bdc5e03-eceb-478f-b924-1dca2b94e3f3)
![image.png](https://images.zenhubusercontent.com/61ae5dab07758299775d9d78/d4c56dde-7e48-4742-aea8-e164486bc5c7)
![image.png](https://images.zenhubusercontent.com/61ae5dab07758299775d9d78/e1ae6df6-69df-4383-91eb-ffd77add59cc)
![image.png](https://images.zenhubusercontent.com/61ae5dab07758299775d9d78/857dc936-5244-4853-b5eb-3edb29d8176a)

**Page Behaviours**

- [ ] .
- [ ] .
- [ ] .

**Validations**

- [ ] .
- [ ] .
- [ ] .

**Actions**

- [ ] .
- [ ] .
- [ ] .

### Not included:
- .
- .

## Implementation Details
- **DB:**
  - Tony's suggestion: Keep it all in the existing status table but update the way we fetch status for analysts and the table definition
    - Add a `visible_to_analyst` option to create a status that is applicant-only
    - Add a check such that you cannot create/update a status such that `visible_to_applicant` and `visible_to_analyst` are both false - we don't need statuses that are invisible
    - Create an analyst-specific computed column where we get the latest status and filter only those that are visible to analysts
    - Create a special applicant-only "Conditionally Approved" and "Received" for this form to switch back and forth from
    - Use that when updating the status in the custom mutation for saving the form (saving the form + changing the status should be one transaction)
- **Middleware:**
  - .
  - .
- **Frontend:**
  - Add check when the value of the status has changed for that modal + initiating the save
  - .
- **Tests:**
  - .
  - .
- **Documentation:**
  - Maybe a documentation update for the nature of statuses? May need some updates with analysts who are querying statuses via metabase
  - .

**Comment by MarsRomer** [comment]
something strange seems afoot... 
The Internal status change doesn't seem to stick. I changed the status to conditionally approved on couple of different applications and when I went to a different page it reverted back.  
The External status change also doesn't seem to work.  I put in all the info needed to change the external status to conditionally approved and clicked save and the modal popped up, I confirmed my decision, and it kept me in edit mode
![image.png](https://images.zenhubusercontent.com/63c8370399512153492c8e97/882782be-1a0f-45c3-a3d1-14614bcf5231)
![image.png](https://images.zenhubusercontent.com/63c8370399512153492c8e97/ea61acf1-9c98-4b51-b30e-9fb061c37385)
After confirming:
![image.png](https://images.zenhubusercontent.com/63c8370399512153492c8e97/6b101e20-dd29-4708-a95b-fde49119fd1b)
